lib/ecto_shorts/common_filters.ex

```diff
@@ -51,36 +51,72 @@ defmodule EctoShorts.CommonFilters do
   CommonFilters.convert_params_to_filter(User, %{name: %{!=: {:lower, "billy"}}})
   ```
   """
+  alias EctoShorts.{CommonSchemas, QueryBuilder}
 
-  alias EctoShorts.QueryBuilder
+  @type query :: EctoShorts.query()
+  @type queryable :: EctoShorts.queryable()
+  @type source :: EctoShorts.source()
+  @type source_queryable :: EctoShorts.source_queryable()
+  @type params :: EctoShorts.params()
+
+  @type filter_key :: EctoShorts.QueryBuilder.filter_key()
+  @type filter_value :: EctoShorts.QueryBuilder.filter_value()
 
   @common_filters QueryBuilder.Common.filters()
 
-  @doc "Converts filter params into a query"
+  @behaviour EctoShorts.QueryBuilder
+
+  @doc """
+  Converts filter params into a query
+  """
   @spec convert_params_to_filter(
-    queryable :: Ecto.Queryable.t(),
-    params :: Keyword.t | map
-  ) :: Ecto.Query.t
-  def convert_params_to_filter(query, params) when params === %{} do
-    query
+    query_or_queryable_or_source_queryable :: query() | queryable() | source_queryable(),
```

Comment:
done.

Reply:
No worries, thanks for taking the time to look at it @BobbieBarker! For the areas that are unclear feel free to raise them and I can provide more information. Ecto is pretty huge in terms of functionality so I think there is opportunities to improve on the documentation as well for how it's supported.

Reply:
Some of this could also be split out into smaller prs as well taking a second look at this. For eg the refactor of the less public api such as the QueryBuilder.Common / QueryBuilder.Schema could be it's own pr.

Reply:
1. No i did not use `git mv`, good catch. I could go back to the last commit on main and then re-play those changes with `git mv` and push it back up. Is that okay?

2. Yes the main goal is to support polymorphic associations as described in the documentation. When developing the feature I didn't want to repeat the code and the handling for the syntax so `EctoShorts.CommonSchemas` was created as interface to handle the translation based on whether `{source, schema}` or `schema` is given.